Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```diff
@@ -52,6 +54,125 @@
     "", ".exe", ".bat"
   }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
 
+  /**
+   * For a given Process ID, kill any child processes and then kill the process. Works on Unix only,
+   * as it leverages pgrep and kill commands.
+   *
+   * @param pid The Process for which to terminate including it's direct children.
+   */
+  public static void killLinuxProcessTree(int pid) {
+    if (pid < 1) {
+      throw new IllegalArgumentException("Process ID should be greater than 0");
+    }
+    // get child process PIDs and kill them
+    getChildUnixProcessPids(pid).ifPresent(ExecUtils::sendSigKill);
+
+    // kill process itself, after all the child processes have been terminated
+    sendSigKill(pid);
+  }
+
+  /**
+   * Runs pgrep -P for a given Process ID, to get a list of child processes as Process IDs.
+   *
+   * @param pid The parent process ID to check for child processes.
+   * @return Optional<String[]> An array of process IDs for the children of pid.
+   */
+  public static Optional<int[]> getChildUnixProcessPids(int pid) {
+    Optional<int[]> pids = Optional.empty();
+    try {
+      Process getChildPid = Runtime.getRuntime().exec("pgrep -P " + pid);
+      getChildPid.waitFor();
+      StringBuilder childPid = new StringBuilder();
+      if (getChildPid.exitValue() == 0) {
+        CharStreams.copy(new InputStreamReader(getChildPid.getInputStream()), childPid);
+      } else {
+        StringBuilder errorOutput = new StringBuilder();
+        CharStreams.copy(new InputStreamReader(getChildPid.getErrorStream()), errorOutput);
+        LOGGER.debug("getChildPid function did not run properly.\n" + errorOutput);
+      }
+      getChildPid.destroy();
+      // convert string to array of ints
+      pids =
+          Optional.of(
+              Arrays.stream(childPid.toString().replaceAll("\n", " ").split(" "))
+                  .mapToInt(Integer::parseInt)
+                  .toArray());
+    } catch (IOException | InterruptedException e) {
+      LOGGER.error("Error getting child processes for: " + pid, e);
+    } catch (NumberFormatException e) {
+      LOGGER.error("Output of getChildPids command not parsable as integers", e);
```

Comment:
Being an `error` level log message, would be good if it was a more useful error message for an administrator.

```suggestion
      LOGGER.error("Unsupported output from 'pgrep -P'. Unable to terminate child processes.");
      LOGGER.error("'pgrep' output: " + childPid.toString());
      LOGGER.error("Parsing exception.", e);
```